crates/net/network/src/transactions/fetcher.rs

```diff
@@ -101,203 +158,309 @@ impl TransactionFetcher {
+                }
+            }
         };
+        let hash = hashes_to_request.first()?;
 
-        if hashes.msg_version().is_eth68() {
-            return self.pack_hashes_eth68(hashes, peer_id)
-        }
-        self.pack_hashes_eth66(hashes)
+        // pop hash that is loaded in request buffer from cache of hashes pending fetch
+        drop(hashes_pending_fetch_iter);
+        _ = self.hashes_pending_fetch.remove(hash);
+
+        idle_peer
     }
 
-    /// Packages hashes for [`GetPooledTxRequest`] up to limit as defined by protocol version 66.
-    /// If necessary, takes hashes from buffer for which peer is listed as fallback peer.
+    /// Packages hashes for a [`GetPooledTxRequest`] up to limit. Returns left over hashes. Takes
+    /// a [`RequestTxHashes`] buffer as parameter for filling with hashes to request.
     ///
     /// Returns left over hashes.
-    pub(super) fn pack_hashes_eth66(&mut self, hashes: &mut ValidTxHashes) -> ValidTxHashes {
-        if hashes.len() <= GET_POOLED_TRANSACTION_SOFT_LIMIT_NUM_HASHES {
-            return ValidTxHashes::empty_eth66()
+    pub(super) fn pack_hashes(
+        &mut self,
+        hashes_to_request: &mut RequestTxHashes,
+        hashes_from_announcement: ValidAnnouncementData,
+    ) -> RequestTxHashes {
+        if hashes_from_announcement.msg_version().is_eth68() {
+            return self.pack_hashes_eth68(hashes_to_request, hashes_from_announcement)
         }
-        let surplus_hashes = hashes.split_off(GET_POOLED_TRANSACTION_SOFT_LIMIT_NUM_HASHES - 1);
-
-        ValidTxHashes::new_eth66(surplus_hashes)
+        self.pack_hashes_eth66(hashes_to_request, hashes_from_announcement)
     }
 
-    /// Evaluates wether or not to include a hash in a `GetPooledTransactions` version eth68
-    /// request, based on the size of the transaction and the accumulated size of the
-    /// corresponding `PooledTransactions` response.
+    /// Packages hashes for a [`GetPooledTxRequest`] from an
+    /// [`Eth68`](reth_eth_wire::EthVersion::Eth68) announcement up to limit as defined by protocol
+    /// version 68. Takes a [`RequestTxHashes`] buffer as parameter for filling with hashes to
+    /// request.
     ///
-    /// Returns `true` if hash is included in request. If there is still space in the respective
-    /// response but not enough for the transaction of given hash, `false` is returned.
-    fn include_eth68_hash(&self, acc_size_response: &mut usize, hash: TxHash) -> bool {
-        debug_assert!(
-            self.eth68_meta.peek(&hash).is_some(),
-            "can't find eth68 metadata for `%hash` that should be of version eth68, broken invariant `@eth68_meta` and `@self`,
-`%hash`: {},
-`@self`: {:?}",
-            hash, self
-        );
+    /// Returns left over hashes.
+    ///
+    /// Loops through hashes passed as parameter and checks if a hash fits in the expected
+    /// response. If no, it's add to surplus hashes. If yes, it's add to hashes to request and
```

Comment:
nit 
```suggestion
    /// response. If no, it's added to surplus hashes. If yes, it's added to hashes to the request and
```

also, reading this fn over, i wasn't aware at first that it returned the surplus hashes, it would be nice to add this in the docs 😊